### PR TITLE
Rewrite cloud

### PR DESCRIPTION
**Pull request by prsauer**

### CODE REVIEW

File: packages/cloud/serverless.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+plugins:
+  - serverless-google-cloudfunctions
+
+package:
+  include:
+    - 'index.js'
+    - 'dist/**'
+
+functions:
```
Comment (kunchenguid) [inline packages/cloud/serverless.yml:19]: should these two be separate or would it be better to make it a single "onLogFileUploaded" handler that does both parts of the work?

File: packages/shared/src/graphql-server/types/index.ts [hunk 2]
```diff
@@ -36,161 +23,101 @@ export interface ApolloContext {
...
+ * * versatilityHealingDone: number;
+ * * versatilityDamageTaken: number;
+ * * armor: number;
+ * * equipment: EquippedItem[];
+ * * interestingAurasJSON: string;
+ * * item28: number;
+ * * item29: number;
+ */
+export interface ICombatantInfoStub
+  extends Pick<CombatantInfo, 'teamId' | 'specId' | 'talents' | 'pvpTalents' | 'personalRating' | 'highestPvpTier'> {}
```
Comment (kunchenguid) [inline packages/shared/src/graphql-server/types/index.ts:57]: oh I didn't know there's this thing. interesting!

File: packages/cloud/PUBLIC_MATCHES.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+## Acronyms
+
+CS - Cloud Storage
+CF - Cloud Function
+FS - Firestore
+
+## Private match architecture (current)
```
[the remark this answers is not in the document]
Comment (prsauer) [reply packages/cloud/PUBLIC_MATCHES.md:7]: yeah I'm deleting this file, these READMEs were super old

File: packages/cloud/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+In /cloud:
+`npm install`
+`npm run build`
+
+`npm run deploy:dev`
+or
+`npm run deploy:prod`
+
+## Manual steps to config env:
```
[the remark this answers is not in the document]
Comment (prsauer) [reply packages/cloud/README.md:21]: yep

File: packages/cloud/README.md [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+
+Create buckets
+`wowarenalogs-log-files-dev` : public read
+`wowarenalogs-log-files-prod` : public read
+
+set cors using cors.json
+
+## Cloud function commands
+
+### Clear temp
```
[the remark this answers is not in the document]
Comment (prsauer) [reply packages/cloud/README.md:31]: good question. I'm going to delete these sections as I can't even remember what I was doing, some of this stuff is close to 2yr old

File: packages/cloud/README.md [hunk 6]
```diff
@@ -0,0 +1,80 @@
+# Cloud Functions
+
+This repo contains cloud functions that handle match ingress, data storage, and log parsing
+
+## Deployment
```
[the remark this answers is not in the document]
Comment (prsauer) [reply packages/cloud/README.md:5]: I hope so, I need to check out what's available to get rm and cp commands on windows+mac

[the remark this answers is not in the document]
Comment (kunchenguid) [reply packages/cloud/README.md:5]: there is a npm package "shx" that does this. super handy

File: packages/desktop/hooks/LocalCombatsContext/index.tsx [hunk 7]
```diff
@@ -44,8 +44,7 @@ export const LocalCombatsContextProvider = (props: IProps) => {
           })
         )
           if (wowVersion === combat.wowVersion) {
-            // TODO: write upload utiltiy
-            // SharedUtils.uploadCombatAsync(combat, userId);
+            uploadCombatAsync(combat, auth.battlenetId);
 
             // console.log('combatMonitorEffect.handleNewCombat', combat);
             // TODO: a more robust way of making sure the handlers only sign up for a single version
```
Comment (kunchenguid) [inline packages/desktop/hooks/LocalCombatsContext/index.tsx:50]: can probably clean up these comments?

File: packages/desktop/hooks/LocalCombatsContext/index.tsx [hunk 8]
```diff
@@ -68,7 +67,7 @@ export const LocalCombatsContextProvider = (props: IProps) => {
         if (wowVersion === combat.wowVersion) {
           console.log('ShuffleEnded');
```
Comment (kunchenguid) [inline packages/desktop/hooks/LocalCombatsContext/index.tsx:68]: do we need these console logs?

**Review by kunchenguid — APPROVED** [review]

**Pull request merged**